presto-main/src/main/resources/webapp/src/components/ResourceGroupView.jsx

```diff
@@ -0,0 +1,349 @@
+                schedulingPolicy: resources.schedulingPolicy,
+                schedulingWeight: resources.schedulingWeight,
+                softMemoryLimit: resources.softMemoryLimit,
+                softConcurrencyLimit: resources.softConcurrencyLimit,
+                hardConcurrencyLimit: resources.hardConcurrencyLimit,
+                maxQueuedQueries: resources.maxQueuedQueries,
+                numEligibleSubGroups: resources.numEligibleSubGroups,
+                workersPerQueryLimit: resources.workersPerQueryLimit,
+                subGroups: resources.subGroups.map(subgrp => subgrp.id.join('.')),
+                runningQueries: resources.runningQueries,
+                numRunningQueries: dataSet.current.numRunningQueries[dataSet.current.numRunningQueries.length - 1],
+                numQueuedQueries: dataSet.current.numQueuedQueries[dataSet.current.numQueuedQueries.length - 1],
+                numAggregatedRunningQueries: dataSet.current.numAggregatedRunningQueries[dataSet.current.numAggregatedRunningQueries.length - 1],
+                numAggregatedQueuedQueries: dataSet.current.numAggregatedQueuedQueries[dataSet.current.numAggregatedQueuedQueries.length - 1],
+                memoryUsage: dataSet.current.memoryUsage[dataSet.current.memoryUsage.length - 1],
+                showDoc: false,
+                showResource: true
+            });
+        }).catch((err) => {
+            console.log(err);
+            setValues({...values, showDoc: true, showResource: false, error: {message: 'There is no Resource Groups settings or data.'}});
+        });
+    }
+
+    function updateCharts() {
+        $('#running-queries-sparkline').sparkline(dataSet.current.numRunningQueries, $.extend({}, SPARKLINE_PROPERTIES, {chartRangeMin: 0}));
+        $('#queued-queries-sparkline').sparkline(dataSet.current.numQueuedQueries, $.extend({}, SPARKLINE_PROPERTIES, {chartRangeMin: 0}));
+        $('#agg-running-queries-sparkline').sparkline(dataSet.current.numAggregatedRunningQueries, $.extend({}, SPARKLINE_PROPERTIES, {chartRangeMin: 0}));
+        $('#agg-queued-queries-sparkline').sparkline(dataSet.current.numAggregatedQueuedQueries, $.extend({}, SPARKLINE_PROPERTIES, {chartRangeMin: 0}));
+        $('#memory-usage-sparkline').sparkline(dataSet.current.memoryUsage, $.extend({}, SPARKLINE_PROPERTIES, {numberFormatter: formatDataSizeBytes}));
+        $('[data-toggle="tooltip"]').tooltip();
+        timerid.current = setTimeout(fetchData, 1000);
+    }
+
+    React.useEffect(() => {
+        if (!timerid.current) {
+            // first call
+            timerid.current = setTimeout(fetchData, 1000);
+        } else if (values.rootGroups.length === 0) {
+            // the following calls
+            updateCharts();
+        }
+        return () => {
+            clearTimeout(timerid.current);
+        };
+    });
+
+    var fullResPath = '';
+    const lastIdx = values.id.length -1;
+    const rsPath = values.id.map((rs, idx) => {
```

Comment:
This variable is not used, you can remove it.

Reply:
From line 259 to 275

Reply:
yes, I refactored this part into a react comp and forgot to remove the code here.